pkg/R/RDD.R

```diff
@@ -1,6 +1,46 @@
 # RDD in R implemented in S4 OO system.
 
-#setOldClass("jobjRef")
+# Maintain a reference count of Java object references
+# This allows us to GC the java object when it is safe
+.validJobjs <- new.env(parent = emptyenv())
+
+# List of object ids to be removed
+.toRemoveJobjs <- new.env(parent = emptyenv())
+
+getJobj <- function(objId) {
+  newObj <- jobj(objId)
+  if (exists(objId, .validJobjs)) {
+    .validJobjs[[objId]] <- .validJobjs[[objId]] + 1
+  } else {
+    .validJobjs[[objId]] <- 1
+  }
+  newObj
+}
+
+# Handler for a java object that exists on the backend.
+jobj <- function(objId) {
+  if (!is.character(objId)) stop("object id must be a character")
+  obj <- structure(new.env(parent = emptyenv()), class = "jobj")
```

Comment:
So the reason we need a new env for a jobj is that we can only register finalizers for environments or external references pointers. Given that we need the finalizer for GC on the Java side, I am not sure we have a better solution for now